### PR TITLE
HAAR-231 Add find address endpoint

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonAddressMaintenanceResource.kt [hunk 1]
```diff
@@ -47,8 +47,7 @@ class PrisonAddressMaintenanceResource(
     responses = [
       ApiResponse(
         responseCode = "200",
-        description = "Address Information Updated",
-        content = [Content(mediaType = "application/json", schema = Schema(implementation = AddressDto::class))]
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonAddressMaintenanceResource.kt:51]: Removed content section (from a previous PR) as not needed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonResource.kt [hunk 2]
```diff
@@ -13,20 +13,23 @@ import org.springframework.http.MediaType
...
 import uk.gov.justice.digital.hmpps.prisonregister.service.PrisonService
 import javax.validation.constraints.Size
 
 @RestController
 @Validated
-class PrisonResource(private val prisonService: PrisonService) {
-  @GetMapping("/$PRISON_BY_ID", produces = [MediaType.APPLICATION_JSON_VALUE])
+@RequestMapping("/prisons", produces = [MediaType.APPLICATION_JSON_VALUE])
+class PrisonResource(private val prisonService: PrisonService, private val addressService: PrisonAddressService) {
+  @GetMapping("/id/{prisonId}")
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonResource.kt:32]: tidy up

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/GpResourceTest.kt [hunk 3]
```diff
@@ -59,7 +59,7 @@ class GpResourceTest : IntegrationTest() {
     webTestClient.get().uri("/gp/prison/1234561234561")
       .exchange()
       .expectStatus().isBadRequest
-      .expectBody().json("prison_id_badrequest_getPrisonFromId".loadJson())
+      .expectBody().json("prison_id_badrequest_getPrisonFromIdForGP".loadJson())
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/GpResourceTest.kt:62]: Needed a separate file as it was using the same json.  This isn't code that our team maintain, so left the json returned as previously expected

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Pull request merged**